### PR TITLE
[DOCS] Replace `put` with `create or update` in API names

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:
* Replaces API names containing `put` with `create`, `update`, or `create or update`
* Updates references to changed API names
* Renames the `Add index alias` API to `Create or update index alias`
* Renames the `Update index alias` API to `Bulk index alias`, similar to the [Bulk document API](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html)
* Rewraps several updated sections
* Simplifies wording in several parts of the Java HLRC docs

### CODE REVIEW

**Comment by jrodewig** [comment]
@lcawl  This PR updates a few ML API names in the Java HLRC docs. Let me know if you'd like me to split those files off into a separate PR. Thanks!

File: docs/java-rest/high-level/rollup/put_job.asciidoc [hunk 1]
```diff
@@ -1,15 +1,15 @@
 [role="xpack"]
 [[java-rest-high-x-pack-rollup-put-job]]
-=== Put Rollup Job API
+=== Create or update rollup job API
 
 experimental::[]
 
-The Put Rollup Job API can be used to create a new Rollup job
+The create or update rollup job API can be used to create a new Rollup job
 in the cluster. The API accepts a `PutRollupJobRequest` object
```
Comment (debadair) [inline docs/java-rest/high-level/rollup/put_job.asciidoc:8]:
```suggestion
Creates a new Rollup job or updates an existing one.
The API accepts a `PutRollupJobRequest` object
```

File: docs/java-rest/high-level/rollup/put_job.asciidoc [hunk 2]
```diff
@@ -126,7 +126,7 @@ include-tagged::{doc-tests}/RollupDocumentationIT.java[x-pack-rollup-put-rollup-
 [[java-rest-high-x-pack-rollup-put-rollup-job-execution]]
 ==== Execution
 
-The Put Rollup Job API can be executed through a `RollupClient`
+The API can be executed through a `RollupClient`
```
Comment (debadair) [inline docs/java-rest/high-level/rollup/put_job.asciidoc:129]: Invoked? You really don't execute APIs. :-)

File: docs/reference/indices/aliases.asciidoc [hunk 3]
```diff
@@ -1,12 +1,14 @@
 [[indices-aliases]]
-=== Update index alias API
+=== Bulk index alias API
 ++++
-<titleabbrev>Update index alias</titleabbrev>
+<titleabbrev>Bulk index alias</titleabbrev>
 ++++
 
-Adds or removes index aliases.
+Adds and removes multiple index aliases n a single request. Also deletes
```
Comment (debadair) [inline docs/reference/indices/aliases.asciidoc:7]:
```suggestion
Adds and removes multiple index aliases in a single request. Also deletes
```

**Review by debadair — APPROVED** [review]
Left some comments & suggestions, but they are mostly additional cleanup that isn't the main focus of this PR. Fixing the typo would be good, but deferring the rest is fine. Thanks for jumping on this so quickly!

File: docs/java-rest/high-level/ccr/put_follow.asciidoc [hunk 4]
```diff
@@ -5,29 +5,29 @@
...
 [id="{upid}-{api}"]
-=== Put Follow API
+=== Create follower API
 
 
 [id="{upid}-{api}-request"]
 ==== Request
 
-The Put Follow API allows creates a follower index and make that index follow a leader index.
+Creates a follower index and makes that index follow a leader index.
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/java-rest/high-level/ccr/put_follow.asciidoc:14]: That's the case for our REST API docs. However, the Java HLRC docs follow their own convention and format.

I agree that we should audit these docs, but that's outside the scope of this PR.

File: docs/java-rest/high-level/ccr/put_follow.asciidoc [hunk 5]
```diff
@@ -5,29 +5,29 @@
...
 
 ["source","java",subs="attributes,callouts,macros"]
 --------------------------------------------------
 include-tagged::{doc-tests-file}[{api}-request]
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/java-rest/high-level/ccr/put_follow.asciidoc:18]: I agree. We should audit the Java HLRC docs as part of a follow-up effort.

File: docs/java-rest/high-level/ml/put-calendar-job.asciidoc [hunk 6]
```diff
@@ -5,14 +5,14 @@
 --
 [role="xpack"]
 [id="{upid}-{api}"]
-=== Put {anomaly-jobs} in calendar API
+=== Create {anomaly-jobs} in calendar API
```
Comment (lcawl) [inline docs/java-rest/high-level/ml/put-calendar-job.asciidoc:8]: Aligning this with the REST API... you're not creating something, you're adding something to the calendar
```suggestion
=== Add {anomaly-jobs} to calendar API
```

File: docs/java-rest/high-level/ml/put-trained-model-alias.asciidoc [hunk 7]
```diff
@@ -5,7 +5,7 @@
 --
 [role="xpack"]
 [id="{upid}-{api}"]
-=== Put trained model alias API
+=== Create trained model alias API
```
Comment (lcawl) [inline docs/java-rest/high-level/ml/put-trained-model-alias.asciidoc:8]:
```suggestion
=== Create or update trained model alias API
```

**Review by lcawl — APPROVED** [review]
Two comments, otherwise ML changes LGTM, thanks!

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Pull request merged**